docs/plugins.md

```diff
@@ -0,0 +1,351 @@
+**DON'T**
+
+> Good job on `alt` attributes
+
+> Fix your headers
+
+#### Audit Descriptions
+
+Write audit descriptions that provide brief context for why the audit is important and link to more detailed guides on how to follow its advice. Markdown links are supported, so use them!
+
+**DO**
+
+> Interactive elements like buttons and links should be large enough (48x48px), and have enough space around them, to be easy enough to tap without overlapping onto other elements. [Learn more](https://developers.google.com/web/fundamentals/accessibility/accessible-styles#multi-device_responsive_design).
+
+> All sites should be protected with HTTPS, even ones that don\'t handle sensitive data. HTTPS prevents intruders from tampering with or passively listening in on the communications between your app and your users, and is a prerequisite for HTTP/2 and many new web platform APIs. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/https).
+
+**DON'T**
+
+> Images need alt attributes.
+
+> 4.8.4.4 Requirements for providing text to act as an alternative for images
+> Except where otherwise specified, the alt attribute.... 10,000 words later... and that is everything you need to know about the `alt` attribute!
+
+### Scoring
+
+1. Weight each audit by its importance.
+1. Differentiate scores within an audit by returning a number _between_ `0` and `1`. Scores greater than `0.9` will be hidden in "Passed Audits" section by default.
+1. Don't inflate scores unnecessarily by marking audits as not applicable. When an audit's advice doesn't apply, simply `return {notApplicable: true}`.
+
+### Common Mistakes
+
+The web is a diverse place, and your plugin will be run on pages you never thought existed. Here are a few things to keep in mind when writing your audit to avoid common bugs. The Lighthouse team has made all of these mistakes below, so you're in good company!
+
+#### Forgetting to Filter
+
+Most audits will have a specific use case in mind that will apply to most elements or requests, but there are corner cases that come up fairly frequently that are easy to forget.
+
+**Examples:**
+
+- Non-network network requests (`blob:`, `data:`, `file:`, etc)
+- Non-script scripts (`type="x-shader/x-vertex"`, `type="application/ld+json"`, etc)
+- Tracking pixel images (images with size 1x1, 0x0, etc)
+
+#### Forgetting to Normalize
+
+Most artifacts will try to represent as truthfully as possible what was observed from the page. When possible, the values are normalized according to the spec as you would access them from the DOM, but typically no transformation beyond this is done. This means that some values will have leading or trailing whitespace, be mixed-case, potentially missing, relative URLs instead of absolute, etc.
+
+**Examples:**
+
+- Header names and values
+- Script `type` values
+- Script `src` values
+
+## Examples
+
+- [Google AdSpeed Insights](https://github.com/googleads/ad-speed-insights) - a well-written, but complex, plugin
```

Comment:
kind of ridiculous, but link to the recipe again? If you were just skimming and skipped to the examples section from the ToC you'd miss the earlier links

Reply:
```suggestion
- [Google AdSpeed Insights](https://github.com/googleads/ad-speed-insights) - a well-written, but complex, plugin
- [Lighthouse Plugin Recipe](./recipes/lighthouse-plugin-example)
```